TEMPLATES/mega-linter.yml

```diff
@@ -40,129 +120,227 @@ jobs:
     name: MegaLinter
     runs-on: ubuntu-latest
 
-    # Give the default GITHUB_TOKEN write permission to commit and push, comment
-    # issues & post new PR; remove the ones you do not need
+    ##
+    # This job's `GITHUB_TOKEN` or your added `PAT` must have the following
+    # permissions enabled to `write` for MegaLinter to function.
+    #
+    # Always aim to provide as few permissions as possible for personal access
+    # tokens. By default, all permissions are set to read only. Write
+    # permissions must be set for the following scopes in the described
+    # circumstances:
+    #
+    #   To have MegaLinter write and update a linting report comment on Pull
+    #   Requests, the `pull-requests` and `issues` permissions must be `write`.
+    #
+    #   To enable `APPLY_FIXES`, the `pull-requests` and `contents` permissions
+    #   must be `write`.
+    #
+    # More info at:
+    # https://docs.github.com/en/actions/security-guides/automatic-token-authentication
+    #
     permissions:
       contents: write
       issues: write
       pull-requests: write
 
     steps:
 
-      # Git Checkout
+      ##
+      # Clone the Repository for linting.
+      #
+      # More info at:
+      # https://github.com/actions/checkout
+      #
       - name: Checkout Code
         uses: actions/checkout@v4
         with:
           token: ${{ secrets.PAT || secrets.GITHUB_TOKEN }}
 
-          # If you use VALIDATE_ALL_CODEBASE = true, you can remove this line to
-          # improve performance
+          ##
```

Comment:
@nvuillam While adding that much documentation, would we also say that some linters use the full checked-out history, like gitleaks?

Reply:
That's a good point... a really elegant way might be to let MegaLinter manage this - create a flag that tells the GitHub Runner whether a full history is required (although we're entering the territory of enough effort to require a new Issue to track, here, I would guess).

In the interim... is there a page we can point users to that describes which linters require branch history to work successfully? I didn't even consider that in my configurations, to date, as all documentation treats this as something the developer elects to do, not is required to do in certain circumstances.

Reply:
I don't know off the top of my head if any other than gitleaks use the history. Probably devskim too since with all its problems with repos and for setup when trying to use it.

But there is the notes in the gitleaks page https://megalinter.io/latest/descriptors/repository_gitleaks/ that explain how gitleaks works, like that it will work on what is checked out when it is ran.